addOns/llm/src/main/java/org/zaproxy/addon/llm/ExtensionLlm.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2025 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.llm;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.AbstractPanel;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.extension.SessionChangedListener;
+import org.parosproxy.paros.model.Session;
+import org.zaproxy.addon.llm.ui.ImportDialog;
+import org.zaproxy.addon.llm.ui.LlmReviewAlertMenu;
+import org.zaproxy.addon.llm.ui.settings.LlmOptionsPanel;
+import org.zaproxy.addon.llm.ui.settings.LlmOptionsParam;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+/**
+ * An extension for ZAP that enables researchers to leverage Language Learning Models (LLMs) to
+ * augment the functionalities of ZAP.
+ *
+ * <p>{@link ExtensionAdaptor} classes are the main entry point for adding/loading functionalities
+ * provided by the add-ons.
+ *
+ * @see #hook(ExtensionHook)
+ */
+public class ExtensionLlm extends ExtensionAdaptor {
+
+    private static final Logger LOGGER = LogManager.getLogger(ExtensionLlm.class);
+
+    public static final String NAME = "ExtensionLlm";
+    protected static final String PREFIX = "llm";
+    private static final String[] ROOT = {};
+
+    private ZapMenuItem menuLLM;
+    private AbstractPanel statusPanel;
+    private ImportDialog importDialog;
+    private LlmReviewAlertMenu llmReviewAlertMenu;
+    private LlmOptionsParam llmOptionsParam;
+    private LlmOptionsPanel llmOptionsPanel;
+
+    public ExtensionLlm() {
+        super(NAME);
+        setI18nPrefix(PREFIX);
+    }
+
+    @Override
+    public void init() {
+        super.init();
+        llmOptionsParam = getOptionsParam();
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        if (hasView()) {
+            getView().getOptionsDialog().addParamPanel(ROOT, getOptionsPanel(), true);
```

Comment:
This is the wrong way to add an options panel.
You should use `extensionHook.getHookView().addOptionPanel(getOptionsPanel());`